flake.nix

```diff
@@ -188,9 +184,24 @@
         (mapAttrs' (name: check: nameValuePair "packages/${name}" check) nonBrokenPkgs)
         // {
           inherit (self.packages.${system}) overview;
-          formatting = treefmtEval.config.build.check self;
+          pre-commit-check = pre-commit-hooks.lib.${system}.run {
+            src = ./.;
+            hooks = {
+              alejandra.enable = true;
+            };
+          };
         };
 
+      devShell.${system} = pkgs.mkShell {
+        inherit (self.checks.${system}.pre-commit-check) shellHook;
+        buildInputs = self.checks.${system}.pre-commit-check.enabledPackages;
+      };
+
+      formatter.${system} = pkgs.writeShellScriptBin "formatter" ''
```

Comment:
Could you please you `pkgs.writeShellApplication`? It's not really an issue at the moment, but once people start adding stuff, this will shellcheck it.

Reply:
Done.